src/Microsoft.OData.Client/DeepInsertSaveResult.cs

```diff
@@ -118,6 +120,53 @@ internal void DeepInsertRequest<T>(T resource)
             }
         }
 
+        /// <summary>
+        /// Begins an asynchronous deep insert request.
+        /// </summary>
+        /// <typeparam name="T">The type of the top-level object to be deep inserted.</typeparam>
+        /// <param name="resource">The top-level object of the type to be deep inserted.</param>
+        internal void BeginDeepInsertRequest<T>(T resource)
+        {
+            if (resource == null)
+            {
+                throw Error.ArgumentNull(nameof(resource));
+            }
+
+            PerRequest peReq = null;
+
+            BuildDescriptorGraph(this.ChangedEntries, true, resource);
+
+            try
+            {
+                ODataRequestMessageWrapper deepInsertRequestMessage = this.GenerateDeepInsertRequest();
+                this.Abortable = deepInsertRequestMessage;
+
+                deepInsertRequestMessage.SetContentLengthHeader();
+                this.perRequest = peReq = new PerRequest();
+                peReq.Request = deepInsertRequestMessage;
+                peReq.RequestContentStream = deepInsertRequestMessage.CachedRequestStream;
+
+                AsyncStateBag asyncStateBag = new AsyncStateBag(peReq);
+
+                this.responseStream = new MemoryStream();
+
+                IAsyncResult asyncResult = BaseAsyncResult.InvokeAsync(deepInsertRequestMessage.BeginGetRequestStream, this.AsyncEndGetRequestStream, asyncStateBag);
+
+                peReq.SetRequestCompletedSynchronously(asyncResult.CompletedSynchronously);
+            }
+            catch (Exception e)
+            {
+                this.HandleFailure(peReq, e);

```

Comment:
Is there any way to simulate a scenario that would get this code block triggered? Does a test exists that verifies the behaviour when this code block is executed?

Reply:
Added